src/test/groovy/org/gradle/android/RoomSchemaLocationKspWorkaroundTest.groovy

```diff
@@ -0,0 +1,207 @@
+package org.gradle.android
+
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.Assume
+import spock.lang.Unroll
+
+import static org.gradle.testkit.runner.TaskOutcome.FROM_CACHE
+import static org.gradle.testkit.runner.TaskOutcome.SUCCESS
+
+@MultiVersionTest
+class RoomSchemaLocationKspWorkaroundTest extends RoomWorkaroundAbstractTest {
+
+    @Unroll
+    def "schemas are generated with Ksp into task-specific directory and are cacheable with kotlin and kapt workers enabled (Android #androidVersion) (Kotlin #kotlinVersion) (Ksp #kspVersion)"() {
+        def kotlinVersionNumber = VersionNumber.parse(kotlinVersion)
+        def kspVersionNumber = VersionNumber.parse(kspVersion)
+        // There are kotlin module version errors when using older versions of kotlin with AGP 7.2.0+ in this configuration
+        Assume.assumeFalse(androidVersion >= VersionNumber.parse("7.2.0-alpha01") && kotlinVersionNumber < VersionNumber.parse("1.5.0"))
```

Comment:
I'm confused about this assertion. We don't support or pass in Kotlin 1.4.x. Or is this about something else?

Reply:
done https://github.com/gradle/android-cache-fix-gradle-plugin/pull/429/commits/2c04c999fdbce11c6a6ff61e19b57b902c34e786